### DISCUSSION TITLE
Any way to import text keyframes from Natron/subtitle file

### DISCUSSION THREAD

**Discussion by hellocatfood** [friction2d/friction#144] [General]
I currently use a combination of [Aegisub](https://github.com/TypesettingTools/Aegisub) and Natron to make animated subtitles for videos. My process is to use Aegisub to do the work of creating the subtitles, export a .srt file, and then import that into Natron for adding text styling and any additional animations.

I'd like to try an use friction2d instead of Natron as it has more options for text and character animation. Is there a way to either import keyframes from Natron or read a .srt file? This would avoid having to redo the subtitles again in friction2d.

**Comment by rodlie** [comment]
Hi, sorry for the late reply.

There is no simple way to do this at the moment.

Currently keyframing/animating the text content in Friction is a bit limited, as it don't support expressions and the UI for entering text has no concept of keyframes (but the backend does), this will be improved for the next version (1.0). 

When we get expressions for text:

Since Friction uses JS for expressions, the best way to get data in is to use JSON (this can easily be exported from Natron with some lines of Python). Example of Natron to Friction keyframes: https://github.com/rodlie/natron-addons/tree/main/addons/transformJSON
